sql/src/main/java/io/crate/metadata/sys/MetricsView.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.metadata.sys;
+
+import io.crate.planner.operators.StatementClassifier;
+import org.HdrHistogram.Histogram;
+
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+public final class MetricsView {
+
+    private final Histogram histogram;
+    private final long sumOfDurations;
+    private final long failedCount;
+    private final StatementClassifier.Classification classification;
+
+    /**
+     * Create a read-only view onto the provided metrics.
+     *
+     * The given histogram must not be modified by whoever creates the MetricsView.
+     */
+    public MetricsView(Histogram histogram,
+                       long sumOfDurations,
+                       long failedCount,
+                       StatementClassifier.Classification classification) {
+        this.histogram = histogram;
```

Comment:
Wouldn't it be safer if the `histogram.copy()` is happening here? Afaik the requirement of a copy is only used here inside.

Reply:
The call-site would still be responsible to safe-guard the histogram. In the specific use-case the CTOR call would have to be moved into the synchronized block. So I don't think anything would be won but it would extend the duration of the synchronized block.